src/java/org/apache/cassandra/config/YamlConfigurationLoader.java

```diff
@@ -119,7 +119,8 @@ public Config loadConfig(URL url) throws ConfigurationException
                 throw new AssertionError(e);
             }
 
-            Constructor constructor = new CustomConstructor(Config.class);
+
+            Constructor constructor = new CustomConstructor(Config.class, Yaml.class.getClassLoader());
```

Comment:
Feel we should use `CustomConstructor.class.getClassLoader()` or `Thread.currentThread().getContextClassLoader()`?